code/modules/power/apc/apc_attack.dm

```diff
@@ -110,13 +110,17 @@
 		return TRUE
 	if(!HAS_SILICON_ACCESS(user))
 		return TRUE
+	. = TRUE
 	var/mob/living/silicon/ai/AI = user
 	var/mob/living/silicon/robot/robot = user
-	if(aidisabled || malfhack && istype(malfai) && ((istype(AI) && (malfai != AI && malfai != AI.parent)) || (istype(robot) && (robot in malfai.connected_robots))))
-		if(!loud)
-			balloon_alert(user, "it's disabled!")
-		return FALSE
-	return TRUE
+	if(istype(AI) || istype(robot))
+		if(istype(malfai) && (malfai != AI || !(robot in malfai.connected_robots)))
+			. = FALSE 
+		if(aidisabled)
+			. = FALSE
```

Comment:
```suggestion
		if(aidisabled)
			. = FALSE
		else if(istype(malfai) && (malfai != AI || !(robot in malfai.connected_robots)))
			. = FALSE 
```
Don't need to bother checking for if the AI is our malfai if AI is disabled regardless

(Yes I know this can be one-lined but I understand the onus behind splitting it for readability)